lib/SILOptimizer/Utils/SILInliner.cpp

```diff
@@ -560,25 +600,11 @@ SILValue SILInlineCloner::borrowFunctionArgument(SILValue callArg,
                                                  FullApplySite AI) {
   if (!AI.getFunction()->hasOwnership()
       || callArg.getOwnershipKind() != ValueOwnershipKind::Owned) {
-    return callArg;
+    return SILValue();
```

Comment:
I did this since I wanted to make it so that the function could fail. This made it so that we eliminated unneeded assigns of callArg -> callArg and also allows me to use the invariant that this is not a borrowed argument if we return SILValue().

This then allowed me to use this failure condition to get just the indices of arguments that need to be borrowed.